custom-config-server-client/src/main/java/com/microsoft/sample/customConfigServerClient/AccessTokenManager.java

```diff
@@ -6,35 +6,53 @@
 
 package com.microsoft.sample.customConfigServerClient;
 
-import com.microsoft.azure.AzureEnvironment;
-import com.microsoft.azure.credentials.ApplicationTokenCredentials;
 import java.io.IOException;
 import java.io.InputStream;
+import java.time.OffsetDateTime;
 import java.util.Properties;
 
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.identity.ClientSecretCredentialBuilder;
+
 public class AccessTokenManager {
 
     static {
         Properties prop = new Properties();
         InputStream in = AccessTokenManager.class.getResourceAsStream("/application.properties");
         try {
             prop.load(in);
-            tokenClientId = prop.getProperty("access.token.clientId");
-            String tenantId = prop.getProperty("access.token.tenantId");
-            String secret = prop.getProperty("access.token.secret");
-            String clientId = prop.getProperty("access.token.clientId");
-            credentials = new ApplicationTokenCredentials(
-                clientId, tenantId, secret, AzureEnvironment.AZURE);
+
+            clientId = prop.getProperty("access.token.clientId");
+            tenantId = prop.getProperty("access.token.tenantId");
+            secret = prop.getProperty("access.token.secret");
+
         } catch (IOException e) {
             System.err.println(e.getMessage());
         }
     }
 
-    private static String tokenClientId;
+    private static String clientId;
 
-    private static ApplicationTokenCredentials credentials;
+    private static String secret;
 
-    public static String getToken() throws IOException {
-        return credentials.getToken(tokenClientId);
+    private static String tenantId;
+
+    private AccessToken token;
+
+    public String getToken() throws IOException {
+        if (token == null || token.getExpiresAt().isBefore(OffsetDateTime.now().plusMinutes(5))) {
+            TokenCredential credential = new ClientSecretCredentialBuilder()
+                    .clientId(clientId)
+                    .clientSecret(secret)
+                    .tenantId(tenantId)
+                    .build();

```

Comment:
This tokenCredential should be a private member of the AccessTokenManager, instead of being local var scope.

Reply:
My thought was the token is private member. Once token is expired, build TokenCredential and get token. Does it make sense?

Reply:
Yes, we could cache the token, but we probably don't need to construct the TokenCredential every time, we only need to construct it in the ctor.